dali/test/python/test_operator_random_bbox_crop.py

```diff
@@ -0,0 +1,238 @@
+# Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import nvidia.dali as dali
+from nvidia.dali.backend_impl import TensorListGPU
+import numpy as np
+import os
+
+class RandomBBoxCropSynthDataPipeline(Pipeline):
+    def __init__(self, device, batch_size,
+                 boxes,
+                 labels = None,
+                 thresholds = [0, 0.01, 0.05, 0.1, 0.15],
+                 scaling = [0.3, 1.0],
+                 aspect_ratio = [0.5, 2.0],
+                 ltrb = True,
+                 num_attempts = 100,
+                 allow_no_crop = False,
+                 input_shape = None,
+                 crop_shape = None,
+                 num_threads=1, device_id=0, num_gpus=1):
+        super(RandomBBoxCropSynthDataPipeline, self).__init__(
+            batch_size, num_threads, device_id, seed=1234)
+        self.device = device
+        self.boxes = boxes
+        self.labels = labels
+        self.boxes_inputs = ops.ExternalSource()
+        self.labels_inputs = ops.ExternalSource()
+
+        self.bbox_crop = ops.RandomBBoxCrop(
+            device = self.device,
+            aspect_ratio = aspect_ratio,
+            scaling = scaling,
+            thresholds = thresholds,
+            ltrb = ltrb,
+            num_attempts = num_attempts,
+            allow_no_crop = allow_no_crop,
+            input_shape = input_shape,
+            crop_shape = crop_shape)
+
+    def define_graph(self):
+        self.boxes_data = self.boxes_inputs()
+        self.labels_data = self.labels_inputs()
+        inputs = [self.boxes_data]
+        if self.labels_data is not None:
+            inputs.append(self.labels_data)
+        return self.bbox_crop(*inputs)
+
+    def iter_setup(self):
+        self.feed_input(self.boxes_data, self.boxes)
+        if self.labels is not None:
+            self.feed_input(self.labels_data, self.labels)
+
+def compare_eps(in1, in2, eps = 1e-6):
+    diff1 = np.abs(in1 - in2)
+    diff2 = np.abs(in2 - in1)
```

Comment:
What for? Isn't it the same, by definition?

Reply:
I copied it from somewhere else to move forward with the testing. I'll use numpy.allclose instead